EIPS/eip-7702.md

```diff
@@ -204,6 +202,10 @@ For most intents and purposes, an account delegated to `0x0` is indistinguishabl
 
 This is not ideal and may be a significant enough concern to impact the overall adoption of the EIP. For these reasons, we have opted to include a mechanism which allow users to restore their EOA to its original pureness.
 
+### Delegation of code execution only
+
+Other code retrieving operations like `EXTCODEHASH` do not automatically follow delegations, they operate on the delegation designator itself. If instead delegations were followed, an account would be able to temporarily masquerade as having a particular codehash, which would break contracts that rely on codehashes as an indicator of possible account behavior. A change of behavior in a contract is currently only possible if its code explicitly allows it (in particular via `DELEGATECALL`), and a change of codehash is only possible in the presence of `SELFDESTRUCT` (and as of Cancun only in the same transaction as contract creation), so choosing to follow delegations in `EXTCODE*` opcodes would have created a new type of account that broke prior assumptions.
```

Comment:
Retrieving/reading seems to be mixed in this PR.

Also, note that it is possible to **reset the account delegation** by authorizing the zero address. It is thus possible to change the code hash, from the empty hash, to the `0xef01` hash, and then back to the empty hash. This is possible also after the first authorization (so in two txs). So this introduces a new way to "change" the code hash (from non-empty to empty, which does not have to be in the same tx).